File: samples/java/SupportChat/Server/Configuration/src/main/java/systems/zlink/samples/supportchat/server/configuration/SampleLocationStore.java
package systems.zlink.samples.supportchat.server.configuration;

import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions;
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore;

public final class SampleLocationStore {
    private SampleLocationStore() {}

    public static ZLinkRedisLocationStore create(SampleTopology topology) {
        SampleTopology.Location location = topology.location();
        return new ZLinkRedisLocationStore(
                new ZLinkRedisLocationOptions()
                        .setConnectionString(location.redisEndpoint())
                        .setKeyPrefix(location.redisKeyPrefix()));
    }
}

File: samples/kotlin/SupportChat/Server/Configuration/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/configuration/SampleLocationStore.kt
package systems.zlink.samples.kotlin.supportchat.server.configuration

import systems.zlink.framework.locations.redis.ZLinkRedisLocationOptions
import systems.zlink.framework.locations.redis.ZLinkRedisLocationStore

object SampleLocationStore {
    fun create(topology: SampleTopology): ZLinkRedisLocationStore =
        ZLinkRedisLocationStore(
            ZLinkRedisLocationOptions()
                .setConnectionString(topology.location().redisEndpoint)
                .setKeyPrefix(topology.location().redisKeyPrefix)
        )
}
